assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { trackEvent } from '@web-stories-wp/tracking';
+import { useEffect, useRef, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+import { ChecklistCard, DefaultFooterText } from '../../checklistCard';
+import { PRIORITY_COPY } from '../constants';
+import { useRegisterCheck } from '../countContext';
+
+export async function getStoryAmpValidationErrors({ link, status }) {
+  if (!link || !['publish', 'future'].includes(status)) {
+    return false;
+  }
+
+  try {
+    const response = await fetch(link);
+    const storyMarkup = await response.text();
+    const { status: markupStatus, errors } =
+      window.amp.validator.validateString(storyMarkup);
+
+    if ('FAIL' !== markupStatus) {
+      return false;
+    }
+    const filteredErrors = errors
+      .filter(({ severity }) => severity === 'ERROR')
+      .filter(({ code, params }) => {
+        // Filter out errors that are covered in other checks
+        // Already covered by metadata checks.
+
+        // Missing story poster
+        if ('MISSING_URL' === code && params?.[0].startsWith('poster')) {
+          return false;
+        }
+
+        // Missing publisher logo
+        if (
+          'MISSING_URL' === code &&
+          params?.[0].startsWith('publisher-logo')
+        ) {
+          return false;
+        }
+        // Missing video posters
+        if ('INVALID_URL_PROTOCOL' === code && params?.[0].startsWith('src')) {
```

Comment:
Hmm video posters are set as `<amp-video poster="...">`, not in a `src` attribute.

Need to check for the `poster` attribute and that it's an `amp-video` node.

Reply:
i thought this might be wrong. So when I take the poster out of a video this is the error I'm getting, nothing else. I'll recreated and include some data.

Reply:
Interesting 🤔

Reply:
ok so i've deleted the featured image from the video in media and i've gutted the poster url and id in the editor dev tools.

```
resource": {
            "type": "video",
            "mimeType": "video/mp4",
            "creationDate": "2021-07-28T17:41:49",
            "src": "http://localhost:8899/wp-content/uploads/2021/07/mitch.mp4",
            "width": 610,
            "height": 514,
            "poster": "",
            "posterId": null,
            "id": 920,
            "length": 1,
            "lengthFormatted": "0:01",
            "title": "mitch",
            "alt": "mitch",
            "sizes": [],
            "local": false,
            "isPlaceholder": false,
            "isOptimized": false,
            "isMuted": false,
            "provider": "local",
            "baseColor": [
              76,
              77,
              97
            ]
          },
```

this is the only error I am getting before filtering out anything:
<img width="464" alt="Screen Shot 2021-07-29 at 11 19 38 AM" src="https://user-images.githubusercontent.com/10720454/127545421-151bb818-167f-481b-919b-d666a24369df.png">